integration/admin-client/src/main/java/org/keycloak/admin/client/resource/RolesResource.java

```diff
@@ -37,6 +39,78 @@
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     List<RoleRepresentation> list();
+    
+    /**
+     * @param fullRepresentation if true, return roles with their attributes
+     * @return A list containing all roles.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    List<RoleRepresentation> list(@QueryParam("full") @DefaultValue("false") boolean fullRepresentation);
+    
+    /**
+     * Get roles by pagination params.
+     * @param search max number of occurrences
+     * @param first index of the first element
+     * @param max max number of occurrences
+     * @return A list containing the slice of all roles.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    List<RoleRepresentation> list(@QueryParam("first") Integer firstResult,
+                                  @QueryParam("max") Integer maxResults);
```

Comment:
Would primitive type and larger scope one `long` type fit better for this kind of cases?

Reply:
Mmmh, actually all the other API with pagination use Integer for firstResult and maxResult so we have to be coherent. Otherwise, long is probable a lot too much for a pagination, also theses fields are not required and could be null that's why we use Integer instead of int.